### ISSUE TITLE
VideoPress: make compatible with Gutenberg

### ISSUE THREAD

**Issue by jeherve** [Automattic/jetpack#8764]
It is currently not possible to upload VideoPress videos when using the new Gutenberg editor. This is something we will need to fix before WordPress 5.0.

pafL3P-cU-p2

**Comment by aldavigdis** [comment]
I just tested this out in wp-admin on my WPCOM standbox and it seems to be working.

**Comment by jeherve** [comment]
It does not seem to work at all for me. WordPress.com and Jetpack have always been quite different on that front, so let me try to summarize:

### Calypso

This still works today. You can upload a video to VideoPress, as long as your site (WordPress.com or Jetpack) has at least a Premium plan. You can upload those videos via 2 interfaces in Calypso:

- The media library view: https://wordpress.com/media/
- The editor: https://wordpress.com/post/

### WP-Admin, WP 4.9.8

In this case, things work differently on WordPress.com and on Jetpack sites:

#### Jetpack

A few years back we develop a solution to overwrite the default media upload when you had a Premium Plan, when the VideoPress module was active, and when you were looking at one of 2 pages:
- The post editor: `https://yoursite.com/wp-admin/post.php`
- The grid view of the media library: `https://yoursite.com/wp-admin/upload.php`

On those 2 pages, when you attempt to upload a video, instead of being uploaded to your site it is uploaded to WordPress.com servers via this WordPress.com REST API endpoint:
https://developer.wordpress.com/docs/api/1.1/post/sites/%24site/media/new/

You can see this in action in this video:
https://videopress.com/v/7zIRTgSO

Note the difference between the 2 media items visible at the end of the video: one is hosted on WordPress.com while the other is hosted on the site itself (and thus is not a VideoPress video).

#### WordPress.com (wp-admin on a WordPress.com site)

This does not use the REST API endpoint as above, but uses an older implementation.

### Gutenberg

This is what concerns us here. Right now, to insert a video on a site via Gutenberg, you can use the video block. That block works as is, but:

- [ ] On WordPress.com, that block needs to be aware of whether the person has a plan to allow them to upload a video to VideoPress, otherwise you run into error messages like that, that don't explain that you could purchase a plan to be able to upload a video. https://videopress.com/v/1Piq5IAs
- [ ] On WordPress.com, if the site has a plan, the block needs to upload to VideoPress and display the video in a VideoPress player. Otherwise you run into the following behaviour: https://videopress.com/v/NzcSReUq Everything seems to work here, when in fact the video was not uploaded to VideoPress, not transcoded into different formats to ensure max compatibility with all browsers and a HD button, and does not use the VideoPress player.
- [ ] On Jetpack, the block needs to be made aware of the VideoPress module and its status (active or not), and uploads the video to VideoPress servers instead of storing the video locally. Otherwise the site owner will not be able to use the feature they paid for, they will just be hosting their videos locally like in this example: https://videopress.com/v/HefSy1Ng

In short, the Video block is not aware of plan status or VideoPress as a whole and just works as a local video block.

I hope this clarifies things a bit.

**Comment by aldavigdis** [comment]
New issue: Uploading does not seem to work at all in Microsoft Edge. Also, when using the block in Edge, a lot of `Unhandled promise rejection undefined` errors are spewn out, which may be a related issue.

**Comment by jeherve** [comment]
> Uploading does not seem to work at all in Microsoft Edge. Also, when using the block in Edge, a lot of Unhandled promise rejection undefined errors are spewn out

👍 It may be good to report this in the Gutenberg repo, it does not look like it's been reported yet:
https://github.com/wordPress/gutenberg/issues?q=is%3Aissue+is%3Aopen+video+label%3A%22%5BBlock%5D+Video%22+sort%3Aupdated-desc

**Comment by aldavigdis** [comment]
Okay. I am putting this ticket aside until https://github.com/Automattic/jetpack/issues/10198 has been finalised. I just asked @roccotripaldi to include a global JS variable or attribute called `wpcom_plan` along with the block availability.

That would facilitate this ticket.

**Comment by mmtr** [comment]
Writing this for just letting you know that https://github.com/Automattic/wp-calypso/pull/2974 should have solved the below 🙂

> On WordPress.com, that block needs to be aware of whether the person has a plan to allow them to upload a video to VideoPress, otherwise you run into error messages like that, that don't explain that you could purchase a plan to be able to upload a video. https://videopress.com/v/1Piq5IAs

**Comment by mmtr** [comment]
Closing this after completing the 2nd phase of the VideoPress extension for the block editor: pafL3P-cU-p2

It'll be shipped as part of the Jetpack 7.1 release (https://github.com/Automattic/wp-calypso/pull/31045).
